### PR TITLE
chore: doc echo apis

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change adds swag (https://github.com/swaggo/swag) to doc echo APIs. This is useful because some APIs cannot be expressed within the constraints of grpc<->grpc-gateway land, e.g. endpoints that return CSV, or generally endpoints that accept or return MIME types other than application/json.

## Test Plan
- [x] build docs, make sure they look good.

## Commentary (optional)
eh.. the makefile change is kinda bad, build proto/swagger needs to be pulled up a level, I guess, since now it depends on stuff in ./master and ./proto but I wanted to get an opinion on this before I do any of that.

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/internal/core.go [hunk 1]
```diff
@@ -187,6 +187,16 @@ func (m *Master) getMasterLogs(c echo.Context) (interface{}, error) {
 	return entries, nil
 }
 
+// @Summary Get an aggregated view of resource allocation during the given time period (CSV).
+// @Tags Cluster
+// @ID get-raw-resource-allocation-csv
+// @Accept  json
+// @Produce  text/csv
+// @Param   start query string true "Start time to get allocations for"
+// @Param   end query string true "End time to get allocations for"
```
Comment (dzhu) [inline master/internal/core.go:196]:
```suggestion
// @Param   timestamp_after query string true "Start time to get allocations for (YYYY-MM-DDTHH:MM:SSZ format)"
// @Param   timestamp_before query string true "End time to get allocations for (YYYY-MM-DDTHH:MM:SSZ format)"
```

File: master/internal/core.go [hunk 2]
```diff
@@ -305,6 +315,16 @@ func (m *Master) fetchAggregatedResourceAllocation(
 	}
 }
 
+// @Summary Get a detailed view of resource allocation during the given time period (CSV).
+// @Tags Cluster
+// @ID get-aggregated-resource-allocation-csv
+// @Produce  text/csv
+// @Param   start_date query string true "Start time to get allocations for"
+// @Param   end_date query string true "End time to get allocations for"
+// @Param   period query string true "Period to aggregate over"
```
Comment (dzhu) [inline master/internal/core.go:324]:
```suggestion
// @Param   start_date query string true "Start time to get allocations for (YYYY-MM-DD format for daily, YYYY-MM format for monthly)"
// @Param   end_date query string true "End time to get allocations for (YYYY-MM-DD format for daily, YYYY-MM format for monthly)"
// @Param   period query string true "Period to aggregate over (RESOURCE_ALLOCATION_AGGREGATION_PERIOD_DAILY or RESOURCE_ALLOCATION_AGGREGATION_PERIOD_MONTHLY)"
```
And maybe put `period` first to introduce the daily/monthly choice?

File: master/internal/core.go [hunk 3]
```diff
@@ -305,6 +315,16 @@ func (m *Master) fetchAggregatedResourceAllocation(
 	}
 }
 
+// @Summary Get a detailed view of resource allocation during the given time period (CSV).
```
Comment (dzhu) [inline master/internal/core.go:318]:
```suggestion
// @Summary Get an aggregated view of resource allocation during the given time period (CSV).
```

File: master/internal/core.go [hunk 4]
```diff
@@ -187,6 +187,16 @@ func (m *Master) getMasterLogs(c echo.Context) (interface{}, error) {
 	return entries, nil
 }
 
+// @Summary Get an aggregated view of resource allocation during the given time period (CSV).
```
Comment (dzhu) [inline master/internal/core.go:190]:
```suggestion
// @Summary Get a detailed view of resource allocation during the given time period (CSV).
```

File: master/internal/core.go [hunk 5]
```diff
@@ -187,6 +187,16 @@ func (m *Master) getMasterLogs(c echo.Context) (interface{}, error) {
...
+// @Tags Cluster
+// @ID get-raw-resource-allocation-csv
+// @Accept  json
+// @Produce  text/csv
+// @Param   start query string true "Start time to get allocations for"
+// @Param   end query string true "End time to get allocations for"
+// @Success 200 {} string "experiment_id,kind,username,labels,slots,start_time,end_time,seconds"
```
Comment (dzhu) [inline master/internal/core.go:197]: This could probably bear to be a bit more verbose ("A CSV file containing the fields...").

**Review by dzhu — APPROVED** [review]
I saw changes to `proto/go.{mod,sum}` that I think need to be committed as well.

File: master/internal/core.go (hunk 5 above)
Comment (stoksc) [reply master/internal/core.go:197]:
```suggestion
// @Success 200 {} string "A CSV file containing the fields experiment_id,kind,username,labels,slots,start_time,end_time,seconds"
```

**Pull request merged**